runtime/src/main/java/org/corfudb/runtime/LogReplicationRoutingQueueListener.java

```diff
@@ -0,0 +1,357 @@
+            return;
+        }
+
+        boolean callbackResult = false;
+        List<Queue.RoutingTableEntryMsg> entries =
+                results.getEntries().entrySet().stream().map(CSEntry ->
+                        (Queue.RoutingTableEntryMsg)CSEntry.getValue().get(0).getPayload()).collect(Collectors.toList());
+        if (snapshotSyncInProgress.get()) {
+            // For routing queue listener, we have only 1 entry inside corfuStreamEntries.
+            callbackResult = processUpdatesInSnapshotSync(entries);
+        } else {
+            callbackResult = processUpdatesInLogEntrySync(entries);
+        }
+        if (callbackResult) {
+            // When the client processed the entry successfully, delete the entry from the routing queue.
+            deleteEntryFromQueue(results);
+        }
+        this.lastProcessedEntryTs = results.getTimestamp();
+    }
+
+    private void deleteEntryFromQueue(CorfuStreamEntries results) {
+        Map<TableSchema, List<CorfuStreamEntry>> entries = results.getEntries();
+        List<CorfuStreamEntry> routingQueueTableEntries =
+                entries.entrySet().stream().filter(
+                            e -> e.getKey().getTableName().equals(LogReplicationUtils.REPLICATED_QUEUE_NAME_PREFIX))
+                        .map(Map.Entry::getValue)
+                        .findFirst()
+                        .get();
+
+        for (CorfuStreamEntry entry : routingQueueTableEntries) {
+            Queue.CorfuGuidMsg key = (Queue.CorfuGuidMsg)entry.getKey();
+            // Only process updates where operation type == UPDATE.
+            if (entry.getOperation() == CorfuStreamEntry.OperationType.UPDATE) {
+                // TODO: Check payload type??
+                //  Queue.RoutingTableEntryMsg payload = (Queue.RoutingTableEntryMsg)entry.getPayload();
+                try (TxnContext tx = corfuStore.txn(namespace)) {
+                    Table<Queue.CorfuGuidMsg, Queue.RoutingTableEntryMsg, Queue.CorfuQueueMetadataMsg> routingQueue;
+                    // TODO: Get the table from CorfuStreamEntries schema or registry table?
+                    routingQueue =
+                            corfuStore.openQueue(namespace, LogReplicationUtils.REPLICATED_QUEUE_NAME_PREFIX,
+                                    Queue.RoutingTableEntryMsg.class,
+                                    TableOptions.builder().schemaOptions(
+                                                    CorfuOptions.SchemaOptions.newBuilder()
+                                                            .addStreamTag(LogReplicationUtils.REPLICATED_QUEUE_TAG)
+                                                            .build())
+                                            .build());
+                    tx.delete(routingQueue, key);
+                    tx.commit();
+                } catch (InvocationTargetException e) {
```

Comment:
can be simplified as follows:
catch (InvocationTargetException | NoSuchMethodException | IllegalAccessException e) {
     ..........
}